src/lox_class.rs

```diff
@@ -80,14 +95,30 @@ impl LoxCallable for LoxClass {
 #[derive(Debug, Clone)]
 pub struct LoxInstance {
     class: LoxClass,
-    fields: HashMap<String, Literal>,
+    fields: IndexMap<String, Literal>,
+}
+
+impl PartialEq for LoxInstance {
+    fn eq(&self, other: &Self) -> bool {
+        self.class == other.class && self.fields == other.fields
```

Comment:
this depends on how you want it to work, but i'm guessing just ptr hash would work fine here